### PR TITLE
Update LICENSE.md

### PR DESCRIPTION
**Pull request by thofma**
Mainly as a starting point for discussion. I trimmed it down, because I don't think that what we link against at runtime matters for the license. I also added the vague phrase "Nemo developers".

CC: @fingolfin

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -1,21 +1,13 @@
...
-As Nemo links dynamically at runtime with various libraries all of whose licenses are compatible with GPL v3+, you may safely distribute Nemo *as a whole* under the terms of that license (see the MPFR or MPIR source trees downloaded as part of the Nemo package installation for copies of this license, or refer to it online here: 
-
-https://www.gnu.org/copyleft/gpl.txt
-
 Individual parts of Nemo may be distributed according to more permissive terms as laid out below:
 
-* The MPIR and MPFR libraries are licensed LGPL v3+. The license texts are distributed with the libraries, which are downloaded as part of the Nemo package installation.
-
-* The Flint library is licensed LGPL v2.1+. The license text is supplied with the library, which is downloaded as part of the Nemo package installation.
+* Parts of the file src/flint/fmpz.jl is licensed with the MIT license (see ZZ.jl for precise conditions of the license).
```
Comment (lgoettgens) [inline LICENSE.md:5]:
```suggestion
* Parts of the file src/flint/fmpz.jl is licensed with the MIT license (see the top of that file for precise conditions of the license).
```
still the old name

File: LICENSE.md [hunk 2]
```diff
@@ -38,9 +30,6 @@ Individual parts of Nemo may be distributed according to more permissive terms a
 > THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 > (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 > OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-External libraries include their own licenses:
-
 - [GMP](https://gmplib.org/manual/Copying.html#Copying)
 - [MPFR](https://www.mpfr.org/mpfr-current/mpfr.html#Copying)
 - [MPIR](http://mpir.org/)
```
Comment (thofma) [inline LICENSE.md:35]:
```suggestion
```

File: LICENSE.md [hunk 3]
```diff
@@ -38,11 +30,5 @@ Individual parts of Nemo may be distributed according to more permissive terms a
 > THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 > (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 > OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-External libraries include their own licenses:
-
-- [GMP](https://gmplib.org/manual/Copying.html#Copying)
-- [MPFR](https://www.mpfr.org/mpfr-current/mpfr.html#Copying)
-- [MPIR](http://mpir.org/)
 - [FLINT](http://flintlib.org/)
```
Comment (thofma) [inline LICENSE.md:33]:
```suggestion
```

File: LICENSE.md [hunk 4]
```diff
@@ -1,21 +1,13 @@
...
-As Nemo links dynamically at runtime with various libraries all of whose licenses are compatible with GPL v3+, you may safely distribute Nemo *as a whole* under the terms of that license (see the MPFR or MPIR source trees downloaded as part of the Nemo package installation for copies of this license, or refer to it online here: 
-
-https://www.gnu.org/copyleft/gpl.txt
-
 Individual parts of Nemo may be distributed according to more permissive terms as laid out below:
 
-* The MPIR and MPFR libraries are licensed LGPL v3+. The license texts are distributed with the libraries, which are downloaded as part of the Nemo package installation.
-
-* The Flint library is licensed LGPL v2.1+. The license text is supplied with the library, which is downloaded as part of the Nemo package installation.
+* Parts of the file src/flint/fmpz.jl is licensed with the MIT license (see the top of that file for precise conditions of the license).
```
Comment (fingolfin) [inline LICENSE.md:5]:
```suggestion
* Parts of the file src/flint/fmpz.jl are licensed under the MIT license (see the top of that file for precise conditions of the license).
```

**Comment by fingolfin** [comment]
> I don't think that what we link against at runtime matters for the license

At first I thought "that can't be right" but after thinking more about I tend to agree...

I'll try to summarize my thought process: Nemo.jl crucially depend on FLINT in a way that would mean that if FLINT was GPL licensed, then Nemo would have to be GPL licensed as well.

However, FLINT is licensed under the LGPL 3+. That means that as long as people can replace a binary blob (here: the `libflint` shared library shipped via JLL) with a modified version of that blob (built from suitable FLINT sources), we are fine. And since this is indeed possible (and in fact JLLs make it relatively convenient), I think we are fin.

That said, it still is relevant for people who want to redistribute a closed source Julia tool that includes Nemo.jl, and hence FLINT_jll, that they need to worry about FLINT_jll. But actually those people need to carefully investigate every transitive dependency (including all JLLs) for their licensing terms anyway.


For JLLs that then means tracking down the license of the wrapped file... Hopefully this all will get easier in the future with BinaryBuilder2, where it should be relatively easy to write an SBOM tool that can automatically collect SPDX identifiers for all involved licenses... but we are not here yet... and frankly this is not my problem, it is problem closed source vendors need to deal with.

**Comment by thofma** [comment]
Whoever is making money with this can hire a lawyer to figure this out. I will merge once green.

**Pull request merged**